channeldb/invoices.go

```diff
@@ -2231,6 +2237,255 @@ func (d *DB) cancelHTLCs(invoices kvdb.RwBucket, invoiceNum []byte,
 	return invoice, nil
 }
 
+// addHTLCs tries to add the htlcs in the given InvoiceUpdateDesc.
+func (d *DB) addHTLCs(invoices, settleIndex, setIDIndex kvdb.RwBucket, //nolint:lll,funlen
+	invoiceNum []byte, invoice *invpkg.Invoice, hash *lntypes.Hash,
+	update *invpkg.InvoiceUpdateDesc) (*invpkg.Invoice, error) {
+
+	var setID *[32]byte
+	invoiceIsAMP := invoice.IsAMP()
```

Comment:
Just thinking out of the box...I wonder if it's easier to create two functions, one for adding htlcs for AMP `addHTLCsAMP`, and the other for MPP, since they have a different logical flow.

Reply:
Incoming in the next PR, I did it this way so this PR did not need to include changes in the tests (easier to review + build knowledge about the related code)

Reply:
> I did it this way so this PR did not need to include changes in the tests

what did you mean by `this way`? probably should review the next PR now...😂

Reply:
Not changing the interface